internal/headerbp/metrics.go

```diff
@@ -0,0 +1,72 @@
+package headerbp
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+
+	"github.com/reddit/baseplate.go/internal/prometheusbpint"
+)
+
+const (
+	clientMethodLabel = "client_method"
+	clientNameLabel   = "client_name"
+	headerNameLabel   = "header_name"
+	rpcTypeLabel      = "rpc_type"
+	serverMethodLabel = "server_method"
+	serviceLabel      = "service_name"
+)
+
+var (
+	clientHeadersRejectedTotal = promauto.With(prometheusbpint.GlobalRegistry).NewCounterVec(prometheus.CounterOpts{
+		Name: "baseplate_client_headers_rejected_total",
```

Comment:
nit: `_rejected_headers_total`, the last word before total should be the unit in plural form ("headers")